cpp/src/TimerThread.h

```diff
@@ -0,0 +1,112 @@
+//-----------------------------------------------------------------------------
+//
+//	TimerThread.h
+//
+//  Timer for scheduling future events
+//
+//	Copyright (c) 2017 h3ctrl <h3ctrl@gmail.com>
+//
+//	SOFTWARE NOTICE AND LICENSE
+//
+//	This file is part of OpenZWave.
+//
+//	OpenZWave is free software: you can redistribute it and/or modify
+//	it under the terms of the GNU Lesser General Public License as published
+//	by the Free Software Foundation, either version 3 of the License,
+//	or (at your option) any later version.
+//
+//	OpenZWave is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//	GNU Lesser General Public License for more details.
+//
+//	You should have received a copy of the GNU Lesser General Public License
+//	along with OpenZWave.  If not, see <http://www.gnu.org/licenses/>.
+//
+//-----------------------------------------------------------------------------
+
+#ifndef _TIMERTHREAD_H_
+#define _TIMERTHREAD_H_
+
+#define USE_TR1
+#ifdef USE_TR1
+#include <tr1/functional>
+#define TT_STDBIND std::tr1::bind
+#define TT_STDFUNCTION std::tr1::function
+#else
+#include <functional>
+#define TT_STDBIND std::bind
+#define TT_STDFUNCTION std::function
+#endif
+
```

Comment:
I'm thinking you can automate the selection of headers etc:
`#if __cplusplus >= 201103L
#include <memory>
using std::shared_ptr;
#else
#include <tr1/memory>
using std::tr1::shared_ptr;
#endif`

should work